### PR TITLE
Add `InSample` resampling strategy

### LINKED ISSUE #967: Add `InSample` resampling strategy
When you want to train on all specified data and test on the same data.

So, 

```julia
e = evaluate(model, X, y, resampling=InSample(), measure=l1)
e.measurement[1]
```

is equivalent to:

```julia
mach = machine(model, X, y) |> fit!
yhat = predict(mach, X)
l2(yhat, y)
```

Which is even more convenient when there is battery measures to be evaluated.

### PR DESCRIPTION
**Pull request by ablaom**
Closes #967

To do post-PR:

- re-export `InSample` from MLJ? (tracked)

### CODE REVIEW

File: src/resampling.jl [hunk 1]
```diff
@@ -345,7 +389,7 @@ end
                                    rng=Random.GLOBAL_RNG)
 
 Stratified cross-validation resampling strategy, for use in
-`evaluate!`, `evaluate` and in tuning. Applies only to classification
+`evaluate!`, `evaluate` and intuning. Applies only to classification
```
Comment (OkonSamuel) [inline src/resampling.jl:392]: Did you really mean "intuning" or this is an error?

**Review by OkonSamuel — APPROVED** [review]
Just a minor comment. Asides that LGTM!!.

File: src/resampling.jl (hunk 1 above)
Comment (ablaom) [reply src/resampling.jl:392]: Good catch, thanks.

**Comment by ablaom** [comment]
Thankyou @OkonSamuel for your review.

**Pull request merged**